app/services/statistics_calculator.rb

```diff
@@ -1,37 +1,86 @@
 class StatisticsCalculator
 
+    module Constants
+        NUMBER_OF_QUESTIONS = 10
+        FULL_PERCENTAGE = 100
+    end
+    
+    module Score
+        A = 90
+        B = 75
+        C = 60
+    end
+    
     class << self 
 
         def success_per_question(results)
             statistics = Array.new;
-            questions = Array.new(10,0);
+            questions = Array.new(Constants::NUMBER_OF_QUESTIONS, 0);
             
             results.each {|result| count_correct_answers_in_result(result.answers, questions)}
             claculate_correct_answers_percentage(questions, results.length(), statistics)
             statistics
+        end
 
+        def user_score(results)
+            statistics = Array.new;
+            results.each {|result| count_correct_answers_for_user(result.answers, result.user, statistics)}
+            statistics
         end
 
         private 
         
         def count_correct_answers_in_result(result_answers, questions)
-            for index in 1..10 do
+            for index in 1..Constants::NUMBER_OF_QUESTIONS do
                 if (result_answers[index.to_s])
                     questions[index-1] += 1
                 end
              end
         end
 
         def claculate_correct_answers_percentage(questions, results_length, statistics)
```

Comment:
Same here - instead of receiving statistics as a param and modifying it inside, just declare it inside the function and return it